tests/integration/test_package_managers.py

```diff
@@ -66,6 +66,36 @@
             ),
             id="gomod_vendored_with_flag",
         ),
+        # Test case checks if vendor folder will be created with dependencies in cloned
+        # source repo deps/ folder in output folder should be empty.
+        pytest.param(
+            utils.TestParameters(
+                repo="https://github.com/cachito-testing/gomod-vendor-check-no-vendor.git",
+                ref="7ba383d5592910edbf7f287d4b5a00c5ababf751",
+                packages=({"path": ".", "type": "gomod"},),
+                check_output_json=True,
+                check_deps_checksums=True,
+                flags=["--gomod-vendor-check"],
```

Comment:
We're missing the check_vendor param

Maybe all the check_* params should be True by default? Or, if they're False but the corresponding file exists, you should get an error?

Reply:
Nice catch, I set checks to True. I would probably stay with checking only expected content.

Reply:
> Or, if they're False but the corresponding file exists, you should get an error?

Or we could go away with the parameters altogether and use only the existence of the files to trigger the checks. The downside is that it would probably look very "magical".

When we add the pip integration tests, we will probably want to extend the TestParameters class to serve specific package managers scenarios, so I think having the params default to `true` looks good for now.

Reply:
Basing it only on file existence would have a similar problem that you can forget the file and the test won't yell at you. I think requiring the check by default is the best :+1: 

Maybe we would want to default check_vendor to False if there are no `gomod` packages, but that's a question for the future